myriad-scheduler/src/main/java/org/apache/myriad/scheduler/fgs/YarnNodeCapacityManager.java

```diff
@@ -196,13 +196,22 @@ protected void handleContainerAllocation(RMNode rmNode) {
    */
   @SuppressWarnings("unchecked")
   public void setNodeCapacity(RMNode rmNode, Resource newCapacity) {
-    rmNode.getTotalCapability().setMemory(newCapacity.getMemory());
-    rmNode.getTotalCapability().setVirtualCores(newCapacity.getVirtualCores());
-    LOGGER.debug("Setting capacity for node {} to {}", rmNode.getHostName(), newCapacity);
-    // updates the scheduler with the new capacity for the NM.
-    // the event is handled by the scheduler asynchronously
-    rmContext.getDispatcher().getEventHandler().handle(new NodeResourceUpdateSchedulerEvent(rmNode, ResourceOption.newInstance(
-        rmNode.getTotalCapability(), RMNode.OVER_COMMIT_TIMEOUT_MILLIS_DEFAULT)));
+    //NOOP prevent YARN warning changing to same size
+    if (!Resources.equals(rmNode.getTotalCapability(), newCapacity)) {
+      rmNode.getTotalCapability().setMemory(newCapacity.getMemory());
+      rmNode.getTotalCapability().setVirtualCores(newCapacity.getVirtualCores());
+      LOGGER.debug("Setting capacity for node {} to {}", rmNode.getHostName(), newCapacity);
+      // updates the scheduler with the new capacity for the NM.
+      // the event is handled by the scheduler asynchronously
+      synchronized (yarnScheduler) {
+        if (yarnScheduler.getSchedulerNode(rmNode.getNodeID()) != null) {
+          yarnScheduler.updateNodeResource(rmNode,
+              ResourceOption.newInstance(rmNode.getTotalCapability(), RMNode.OVER_COMMIT_TIMEOUT_MILLIS_DEFAULT));
+        } else {
+          LOGGER.info("Yarn Scheduler doesn't have node {}, probably UNHEALTHY", rmNode.getNodeID());
```

Comment:
I'm not entirely sure why you decided to go with the updateNodeResource method rather than scheduling a new NodeResourceUpdateSchedulerEvent, but from the looks of it they do relatively the same thing in theory.

That said, the comments look wrong, as now the event is handled by the scheduler synchronously, and the javadoc needs to be updated. Additionally the NodeResourceUpdateSchedulerEvent no longer needs to be imported.